Rubberduck.Parsing/ComReflection/DeclarationsFromComProjectLoader.cs

```diff
@@ -0,0 +1,307 @@
+                declarations.AddRange(moduleDeclarations);
+            }
+
+            return declarations;
+        }
+
+        private static ICollection<Declaration> GetDeclarationsForModule(IComType module, QualifiedModuleName moduleName, ProjectDeclaration project)
+        {
+            var declarations = new List<Declaration>();
+
+            var attributes = GetModuleAttributes(module);
+            var moduleDeclaration = CreateModuleDeclaration(module, moduleName, project, attributes);
+            declarations.Add(moduleDeclaration);
+
+            switch (module)
+            {
+                case IComTypeWithMembers membered:
+                    var (memberDeclarations, defaultMember) =
+                        GetDeclarationsForProperties(membered.Properties, moduleName, moduleDeclaration);
+                    declarations.AddRange(memberDeclarations);
+                    AssignDefaultMember(moduleDeclaration, defaultMember);
+
+                    (memberDeclarations, defaultMember) = GetDeclarationsForMembers(membered.Members, moduleName,
+                        moduleDeclaration, membered.DefaultMember);
+                    declarations.AddRange(memberDeclarations);
+                    AssignDefaultMember(moduleDeclaration, defaultMember);
+
+                    if (membered is ComCoClass coClass)
+                    {
+                        (memberDeclarations, defaultMember) = GetDeclarationsForMembers(coClass.SourceMembers,
+                            moduleName, moduleDeclaration, coClass.DefaultMember, true);
+                        declarations.AddRange(memberDeclarations);
+                        AssignDefaultMember(moduleDeclaration, defaultMember);
+                    }
+
+                    break;
+                case ComEnumeration enumeration:
+                    {
+                        var enumDeclaration = new Declaration(enumeration, moduleDeclaration, moduleName);
+                        declarations.Add(enumDeclaration);
+                        var members = enumeration.Members
+                            .Select(enumMember => new ValuedDeclaration(enumMember, enumDeclaration, moduleName))
+                            .ToList();
+                        declarations.AddRange(members);
+                        break;
+                    }
+                case ComStruct structure:
+                    {
+                        var typeDeclaration = new Declaration(structure, moduleDeclaration, moduleName);
+                        declarations.Add(typeDeclaration);
+                        var members = structure.Fields
+                            .Select(f => new Declaration(f, typeDeclaration, moduleName))
+                            .ToList();
+                        declarations.AddRange(members);
+                        break;
+                    }
+            }
+
+            if (module is IComTypeWithFields fields && fields.Fields.Any())
```

Comment:
it shold be possible to put this into the select block as well.

```
case IComTypeWithFields fields when fields.Fields.Any():
   [...]
```

Reply:
I think the reason why this is separate is because this condition is kind of orthogonal to the ones in the select block.